lib/membrane_dashboard/charts/helpers.ex

```diff
@@ -55,43 +73,133 @@ defmodule Membrane.Dashboard.Charts.Helpers do
+  - old data - 2D list contatining metric data before update
+  - paths - list of all paths that will be present in the new data
+
+  Returns list of tuples `{path, data}`, where `path` is pipeline element path and data is a list with values (one value for every timestamp in `interval`). 
+  Data is altered in the way that every non-nil value is a number of processed metric events from the beginning of live update.
+  """
+  @spec to_series(postgrex_result_rows_t(), [float()], atom(), [[non_neg_integer()]], [
+          String.t()
+        ]) :: [{String.t(), [non_neg_integer()]}]
+  def to_series(rows, interval, mode, old_data \\ [], paths \\ []) do
+    data_by_paths = rows_to_data_by_paths(rows)
+
+    initial_accumulators =
+      case mode do
+        :full ->
+          data_by_paths
+          |> Enum.map(fn {path, _data} -> {path, 0} end)
+          |> Enum.into(%{})
+
+        :update ->
+          get_initial_accumulators(old_data, paths)
+      end
+
+    data_by_paths_to_series(data_by_paths, interval, initial_accumulators)
+  end
+
+  # converts rows from `measurements` table to list of tuples `{path, data}`, where data is a list of tuples contatining timestamps and values
+  defp rows_to_data_by_paths(rows) do
+    Enum.group_by(rows, fn [_time, path, _value] -> path end, fn [time, _path, value] ->
+      {time, value}
     end)
+  end
+
+  # converts every `data` from list of tuples `{path, data}` to list of integers (value for every timestamp from `interval`)
+  # if `initial_accumulators` is not `nil`, data is altered in the way that every non-nil value is a number of processed metric events from the beginning of live update
+  # `initial_accumulators` is a map %{path => initial number of processed metrics}
+  defp data_by_paths_to_series(data_by_paths, interval, initial_accumulators \\ nil) do
+    data_by_paths
     |> Enum.map(fn {path, data} ->
       data =
         data
         |> group_by_time()
         |> get_max_value_for_every_timestamp()
         |> Enum.into(%{})
-        |> fill_with_nils(interval)
 
-      {path, data}
+      case initial_accumulators do
+        nil -> {path, fill_with_nils(data, interval)}
+        _ -> {path, fill_with_nils(data, interval, initial_accumulators[path])}
+      end
     end)
   end
 
+  # extracts initial numbers of processed metrics for update data from `old_data`
+  defp get_initial_accumulators([_interval | old_data], paths) do
+    accumulators =
+      old_data
+      |> Enum.map(&extract_max(&1))
+
+    new_paths_count = length(paths) - length(accumulators)
+
+    accumulators = accumulators ++ for _i <- 1..new_paths_count, do: 0
+
+    Enum.zip(paths, accumulators)
+    |> Enum.into(%{})
+  end
+
+  # performs `Enum.reduce/3` to get max value from a list while ignoring all `nils`
+  defp extract_max(path_data) do
+    Enum.reduce(
```

Comment:
```suggestion
    path_data |> Enum.reject(& is_nil(&1)) |> Enum.max()
```
Or if you don't want to create another list and then taking a value out you can use `Enum.max` and pass custom comparator function.

Reply:
`Enum.reject` looks good